static/js/tools/choropleth/choropleth.tsx

```diff
@@ -333,13 +389,47 @@ class ChoroplethMap extends Component {
           .attr("font-weight", "bold")
           .text(title)
       );
+
+      const dateComponent = 
+    
+      ReactDOM.render(this.datePicker(), document.getElementById('date'));
+
+  };
+
+  changeDate = (event) => {
```

Comment:
Please add function comments for all new functions

Reply:
Fixed.